### PR TITLE
Pass X-Okapi-Request-Id MODINV-156

### PR DESCRIPTION
**Pull request by adamdickmeiss**
I can't believe how many classes I had to touch for this. There should
be an "OkapiClient" that just takes context from outer request
and relay headers as necessary.. Now the Okapi headers definitions
are duplicated.

### CODE REVIEW

File: src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java [hunk 1]
```diff
@@ -150,6 +154,10 @@ private void okapiHeaders(HttpClientRequest request) {
       request.headers().add(TOKEN_HEADER, this.token);
     }
 
+    if (this.requestId != null) {
```
Comment (julianladisch) [inline src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java]: This is inconsistent, it should use the same logic as the token above:

    if (StringUtils.isNotBlank(this.requestId)) {

File: src/main/java/org/folio/inventory/common/WebContext.java [hunk 2]
```diff
@@ -36,6 +37,11 @@ public String getUserId() {
     return getHeader(OKAPI_USER_ID_HEADER, "");
   }
 
+  @Override
+  public String getRequestId() {
+    return getHeader(OKAPI_REQUEST_ID);
```
Comment (julianladisch) [inline src/main/java/org/folio/inventory/common/WebContext.java]: This is inconsistent. All other methods return "" if undefined. Can you change to

    return getHeader(OKAPI_REQUEST_ID, "");

File: src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java]: I don't agree. This code is faster and uses a very common idiom.. Null value for absent header.

File: src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java [hunk 3]
```diff
@@ -34,12 +35,21 @@
...
+    throws MalformedURLException {
+
+    this(httpClient, new URL(context.getOkapiLocation()),
+      context.getTenantId(), context.getToken(), context.getUserId(),
+      context.getRequestId(), exceptionHandler);
+  }
+
+  public OkapiHttpClient(HttpClient httpClient,
+    URL okapiUrl,
+    String tenantId,
```
Comment (julianladisch) [inline src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java]: Can you add javadoc that null, an empty ("") String or a whitespace only means no tenantId header?

File: src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java [hunk 4]
```diff
@@ -34,12 +35,21 @@
...
+    String token,
```
Comment (julianladisch) [inline src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java]: Can you add javadoc that null, an empty ("") String or a whitespace only means no token header?

File: src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java [hunk 5]
```diff
@@ -34,12 +35,21 @@
...
+    String userId,
+    String requestId,
```
Comment (julianladisch) [inline src/main/java/org/folio/inventory/support/http/client/OkapiHttpClient.java]: Can you add javadoc null means no requestId header and that this is different from the tenantId and token parameters?

**Pull request merged**